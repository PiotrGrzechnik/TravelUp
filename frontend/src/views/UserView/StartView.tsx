import React from 'react'
import { Card } from 'antd'

type StartViewProps = {
  user?: {
    name?: string
  }
}

const StartView: React.FC<StartViewProps> = props => {
  const { user } = props
  const username = user && user.name
  return <div>Welcome {username}!</div>
}

export default StartView
