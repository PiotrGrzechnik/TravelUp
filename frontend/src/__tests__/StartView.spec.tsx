import React from 'react'
import { shallow } from 'enzyme'

import StartView from 'src/views/UserView/StartView'

describe('StartView', () => {
  test('renders component without crashing', () => {
    const app = shallow(<StartView />)
    expect(app).toMatchSnapshot()
  })

  test('pass props correctly', () => {
    const app = shallow(<StartView user={{ name: 'Mary' }} />)
    expect(app).toMatchSnapshot()
  })

  test('render title text properly', () => {
    const wrapper = shallow(<StartView user={{ name: 'Mary' }} />)
    expect(wrapper.html().includes('Hello, Mary')).toBe(true)
  })

  test('render days if createdAt prop is passed', () => {
    const wrapper = shallow(<StartView user={{ createdAt: '01 01 2020' }} />)
    expect(wrapper.html().includes('days')).toBe(true)
  })
})
