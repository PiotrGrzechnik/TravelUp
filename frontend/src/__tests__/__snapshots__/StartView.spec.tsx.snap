// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartView pass props correctly 1`] = `
<styled.div>
  <Styled(Card)
    title={
      <styled.div>
        Hello, 
        Mary
        !
      </styled.div>
    }
  />
</styled.div>
`;

exports[`StartView renders component without crashing 1`] = `
<styled.div>
  <Styled(Card)
    title={
      <styled.div>
        Hello, 
        !
      </styled.div>
    }
  />
</styled.div>
`;
